### DISCUSSION TITLE
allglyphs and unicode

### DISCUSSION THREAD

**Discussion by Crisium** [Chlumsky/msdf-atlas-gen#100] [Q&A]
using the -allglyphs parameter it does generate all the characters, but the json format seems to be different now in version 1.3
now instead of a key called "unicode" we now have "index" and it does not seem relate to the unicode value?

am I using this wrong, or is something broken here?

**Comment by Chlumsky** [comment]
There are two different modes for indexing glyphs. The default is Unicode, which requires that you supply a character set if you want anything other than ASCII. In this mode, you will get the Unicode values in the outputs (JSON etc.). However, if you use the glyph mode (-glyphset, -glyphs, -allglyphs), then the glyphs are instead indexed by their ordinal numbers in the font file and there will be no Unicode values. This is because there is no simple mapping between Unicode charaters and font glyphs (they can represent ligatures, parts of characters, characters can have multiple visual versions depending on context, glyphs can be shared by multiple characters, etc.).

You can use for example HarfBuzz to find out which glyphs to draw for a given Unicode string.
